lib/puppet/type/puppet_authorization_hocon_rule.rb

```diff
@@ -25,6 +25,31 @@
       unless val.is_a?(Hash)
         raise "Value must be a hash but was #{value.class}"
       end
+      validate_acl(val)
+    end
+
+    def validate_acl(val)
+      ["allow", "deny"].each do |rule|
+        if val.has_key?(rule)
```

Comment:
I think the data types in rule.pp should take care of that case for us (https://github.com/puppetlabs/puppetlabs-puppet_authorization/pull/21/files/e366dcc4b3531608878d4a784599b1878273d0c2#diff-57780fa042c02a1b6068fb93b67fd45dR7). At this point we should only see:

- arrays of hashes and/or strings
- a hash
- a string